jenkins/artifact-signer/Jenkinsfile

```
@@ -0,0 +1,72 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
+            alwaysPull true
+        }
+    }
+    parameters{
+        string(
+                name: 'ARTIFACT_PATH',
+                trim: true,
+                description: 'Path to the artifact or the artifact directory on s3'
+        )
+        choice(
+                choices: ['linux'],
+                name: 'DISTRIBUTION_PLATFORM',
+                description: 'What platform is this distribution build for?'
+        )
+        choice(
+                choices: ['.sig'],
+                name: 'SIGNATURE_TYPE',
+                description: 'What is signature file type?'
+        )
+    }
+    stages {
+        stage('sign') {
+            steps {
+                script {
+                    if (ARTIFACT_PATH == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('ARTIFACT_PATH is not set')
+                    }
+                    ARTIFACT_PATH = ARTIFACT_PATH.replaceAll('/$', "")
+                    set_up()
+                    argsMap = [:]
+                    argsMap['artifactPath'] = "$WORKSPACE/artifacts/${ARTIFACT_PATH}"
+                    argsMap['signatureType'] = "${SIGNATURE_TYPE}"
+                    argsMap['distributionPlatform'] = "${DISTRIBUTION_PLATFORM}"
+                    signArtifacts(argsMap)
```

Pass as parameters, no need to make a map

```suggestion
                    signArtifacts(
                       artifactPath: "$WORKSPACE/artifacts/${ARTIFACT_PATH}"
                       signatureType: SIGNATURE_TYPE
                       distributionPlatform: DISTRIBUTION_PLATFORM
```

---

Agree here.

---

We need a job for standalone artifacts, since we have to do them manually as of now.